test/server/cards/03_TWI/events/OnTheDoorstep.spec.ts

```diff
@@ -0,0 +1,22 @@
+describe('On the Doorstep', function () {
+    integration(function (contextRef) {
+        it('On the Doorstep\'s ability should create three Battle Droid tokens for the controller and ready them', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['on-the-doorstep'],
+                }
+            });
+
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.onTheDoorstep);
+
+            const battleDroids = context.player1.findCardsByName('battle-droid');
+            expect(battleDroids.length).toBe(3);
+            expect(battleDroids).toAllBeInZone('groundArena');
+            expect(battleDroids.every((battleDroid) => !battleDroid.exhausted)).toBeTrue();
```

Comment:
why adding ! on exhausted and not check tobefalse ?

Reply:
That would be the wrong logic. Checking "every unit is exhausted === false" is just checking if at least one unit is ready. But we need to check if all units are ready